lib/Auth/index.js

```diff
@@ -204,6 +204,19 @@ class Auth {
           opts.headers.authorization = `Bearer ${clientToken}`;
         }
 
+        if (stream) {
```

Comment:
We can create a superagent object with the common fields between streaming and no stream request.

```js
const req = superagent(method, url)
  .set(opts.headers)
  .timeout( .... )
  .query(opts.query)
  .send(opts.payload)

if (opts.stream) {
  return req.pipe(opts.stream)
}
begin = new Date().getTime();
({ status, body } = await req());
```

Reply:
Moved stream to opts

Reply:
req isn't a function here. changed to use send  in pipe and await